src/Compilers/CSharp/Portable/Binder/Binder_Operators.cs

```diff
@@ -3504,6 +3505,53 @@ private BoundExpression BindNullCoalescingOperator(BinaryExpressionSyntax node,
             return GenerateNullCoalescingBadBinaryOpsError(node, leftOperand, rightOperand, Conversion.NoConversion, diagnostics);
         }
 
+
+        private BoundExpression BindNullCoalescingAssignmentOperator(AssignmentExpressionSyntax node, DiagnosticBag diagnostics)
+        {
+            BoundExpression leftOperand = BindValue(node.Left, diagnostics, BindValueKind.CompoundAssignment);
+            BoundExpression rightOperand = BindValue(node.Right, diagnostics, BindValueKind.RValue);
+
+            // If either operand is bad, bail out preventing more cascading errors
+            if (leftOperand.HasAnyErrors || rightOperand.HasAnyErrors)
+            {
+                return new BoundNullCoalesingAssignmentOperator(node, leftOperand, rightOperand, CreateErrorType(), hasErrors: true);
+            }
+
+            // Unlike a standard null coalescing expression, the resulting type of the expression a ??= b
+            // must be A (where A is the type of a), as it is where the result of the assignment will end up. Therefore,
+            // we must ensure that B (where B is the type B) is implicitly convertible to A or A0 (where A0 is the
+            // underlying non-nullable type of A).
+            TypeSymbol leftType = leftOperand.Type;
+            Debug.Assert(leftType != null);
+
+            // If A is not a nullable type or reference type, a compile-time error occurs
+            if (!leftType.IsReferenceType && !leftType.IsNullableType())
+            {
+                return GenerateNullCoalescingAssignmentBadBinaryOpsError(node, leftOperand, rightOperand, diagnostics);
+            }
+
+            // If implicit conversion exists from B to A, we store that conversion. At runtime, a is first evaluated. If
+            // a is not null, b is not evaluated. If a is null, b is evaluated and converted to type A, and is stored in a.
+            HashSet<DiagnosticInfo> useSiteDiagnostics = null;
+            var rightConversion = Conversions.ClassifyImplicitConversionFromExpression(rightOperand, leftType, ref useSiteDiagnostics);
+            diagnostics.Add(node, useSiteDiagnostics);
```

Comment:
Do you have a test where there is a use-site diagnostic on the conversion?

Reply:
Will do in a follow-up testing PR.

---
In reply to: [200787304](https://github.com/dotnet/roslyn/pull/28253#discussion_r200787304) [](ancestors = 200787304)